packages/query-persist-client-core/src/createPersister.ts

```diff
@@ -0,0 +1,147 @@
+import { hashKey } from '@tanstack/query-core'
+import type {
+  QueryClient,
+  QueryFunctionContext,
+  QueryKey,
+  QueryState,
+} from '@tanstack/query-core'
+
+export interface PersistedQuery {
+  buster: string
+  timestamp: number
+  queryHash: string
+  queryKey: QueryKey
+  state: QueryState
+}
+
+export interface AsyncStorage {
+  getItem: (key: string) => Promise<string | undefined | null>
+  setItem: (key: string, value: string) => Promise<void>
+  removeItem: (key: string) => Promise<void>
+}
```

Comment:
we already have this interface define in `query-async-storage-persister`:

https://github.com/TanStack/query/blob/1d3ee672d2a1536e3463243813c91902aa97d973/packages/query-async-storage-persister/src/index.ts#L8-L12

I think the best way forward would be to have the async-storage-persister package re-use the one from the core you've defined here ?